### ISSUE TITLE
Correct for gradient non-linearity distortions on the SNR maps

### ISSUE THREAD

**Issue by jcohenadad** [spinal-cord-7t/coil-qc-code#2]

**Comment by evaalonsoortiz** [comment]
@dpapp86 According to our discussions today, I think this is no longer available. If that's the case I'll close this issue.

**Comment by evaalonsoortiz** [comment]
Sorry, I meant to write "needed". Non-distortion correction is not needed.

**Comment by jcohenadad** [comment]
right-- i suppose it would be good to also document why it is not needed-- is it because the quantification of SNR is done on the voxel space of the SNR image, and that the SNR and cord segmentation (used to extract SNR on a slice-wise basis) are not affected by those distortions?

**Comment by evaalonsoortiz** [comment]
After looking at the protocol PDFs I realize that I might have misunderstood @dpapp86. I had understood that distortion correction was turned off for all scans. Looking here https://github.com/spinal-cord-7t/coil-qa-protocol/blob/main/MNI/acdc_7t_spine_073_CoilQA.pdf and here https://docs.google.com/document/d/1rfxDTc-coK96GMKsx5_sBK_srCyuKY7BY51qRYrDmig/edit#heading=h.iytyrwkr1whn I see that distortion correction is ON for all scans EXCEPT coilQA scans (SNR, g-factor, etc). The segmentation is done on the MP2RAGE which is distortion corrected, and that segmentation is used for coilQA images. Therefore, we should visually check if the distortion warrants correction. 

Things to consider:
- The segmentation is expanded along the cord, so the distortion would have to be significant for the cord mask to not incorporate the cord in the coilQA data. 
- At one point we will likely choose to restrict analyses to spinal cord levels (ex: C2-T1) so that we are comparing the same levels across coils that may not have the same FOV. When we do that, it may be that the levels that are distorted in the coilQA images are no longer part of the analyses.

**Comment by jcohenadad** [comment]
Beyond distortion, subject motion could also be cause for mismatch between the MP2RAGE-based mask and the coilQA scans. I would recommend to incorporate in the analysis code a QC output that overlays the mask (used by `sct_extract_metric`) onto the coilQA scans. I'm happy to help with that once the code is on GitHub and ready to be commented.

**Comment by evaalonsoortiz** [comment]
The coilQA data is co-registered to the MP2RAGE so that the SC ROI can be overlaid. At the moment, the SNR along the SC ROI is plotted, but there is no visual QA as you described to check the SC ROI on the SNR image itself. This would be a useful addition. The notebook is now on Github BTW (see: https://github.com/spinal-cord-7t/coil-qc-code/blob/main/THS%20CoilQA%20code%20(Kyota).ipynb)
